packages/shylu/shylu_node/tacho/src/impl/Tacho_Util.hpp

```diff
@@ -193,7 +193,7 @@ template <size_t BufSize, typename SpaceType = Kokkos::DefaultExecutionSpace> st
   void init(value_type &update) { update = 0; }
 
   KOKKOS_INLINE_FUNCTION
-  void join(volatile value_type &update, const volatile value_type &input) { update += input; }
+  void join(volatile value_type &update, const volatile value_type &input) { update = update + input; }
```

Comment:
I think the overload using volatile can be removed altogether (kokkos no longer requires this), though you can test and try that in a separate PR if you prefer